gsy_e_sdk/setups/grid_operator_api_scripts/redis_aziiz_strategy.py

```diff
@@ -0,0 +1,132 @@
+# flake8: noqa
+# pylint: disable=duplicate-code
+"""
+Template file to implement Aziiz grid fees
+strategy through the gsy-e-sdk api client using Redis.
+"""
+import os
+import json
+from time import sleep
+from gsy_e_sdk.redis_aggregator import RedisAggregator
+from gsy_e_sdk.redis_market import RedisMarketClient
+from gsy_e_sdk.utils import log_grid_fees_information
+
+module_dir = os.path.dirname(__file__)
+
+# List of markets' names to be connected with the API
+MARKET_NAMES = [
+    "Grid",
+    "Community",
+]
+
+ORACLE_NAME = "dso"
+SLOT_LENGTH = 15
+MOVING_AVERAGE_PEAK = True
+LOOK_BACK_INDEX = 4
+
+
+class Oracle(RedisAggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+        self.import_kwh = {}
+        self.export_kwh = {}
+        self.balance_hist = []
+        self.balance = {}
+        self.dso_stats_response = {}
+
+    def on_market_cycle(self, market_info):
+        current_market_fee = {}
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            if area_dict["area_name"] in MARKET_NAMES:
+                self.add_to_batch_commands.last_market_dso_stats(area_uuid)
+                current_market_fee[area_dict["area_name"]] = area_dict[
+                    "current_market_fee"
+                ]
+        self.calculate_import_export_balance()
+        next_market_fee = self._set_new_market_fee()
+        log_grid_fees_information(MARKET_NAMES, current_market_fee, next_market_fee)
+
+    def calculate_import_export_balance(self):
+        """Calculate the balance (import - export) for each market."""
+        self.dso_stats_response = self.execute_batch_commands()
+        for market_event in self.dso_stats_response["responses"].items():
+            self.import_kwh[market_event[1][0]["name"]] = market_event[1][0][
+                "market_stats"
+            ]["area_throughput"]["import"]
+            self.export_kwh[market_event[1][0]["name"]] = market_event[1][0][
+                "market_stats"
+            ]["area_throughput"]["export"]
+            self.balance[market_event[1][0]["name"]] = (
+                self.import_kwh[market_event[1][0]["name"]]
+                - self.export_kwh[market_event[1][0]["name"]]
+            )
+        self.balance_hist.append(self.balance)
+
+    def _set_new_market_fee(self):
+        """Return the market fees for each market for the next time slot."""
```

Comment:
General comment: I think we need more docstrings to explain what this strategy is doing. Maybe adding it in the module docstrings might work, or somewhere else. But looking at this file I have the feeling that there's some context information missing.